packages/protocol/test/mocks/MockERC20Vault.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+//
+//   Email: security@taiko.xyz
+//   Website: https://taiko.xyz
+//   GitHub: https://github.com/taikoxyz
+//   Discord: https://discord.gg/taikoxyz
+//   Twitter: https://twitter.com/taikoxyz
+//   Blog: https://mirror.xyz/labs.taiko.eth
+//   Youtube: https://www.youtube.com/@taikoxyz
+
+pragma solidity 0.8.24;
+
+import "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import { BridgedERC20 } from "../../contracts/tokenvault/BridgedERC20.sol";
+
+/// @title Mock ERC20 Vault
+/// @notice This contract manages an ERC20 token vault designed for storing and approving tokens for
+/// airdrops,
+/// as well as allowing the owner to withdraw tokens.
+/// @dev The contract uses OwnableUpgradeable for access control and interacts with BridgedERC20
+/// tokens.
+contract MockERC20Vault is OwnableUpgradeable {
```

Comment:
You don't even need this contract, the vault can be an EOA.

Reply:
EOA just has to approve the airdrop contract